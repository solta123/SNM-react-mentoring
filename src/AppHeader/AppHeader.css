.root {
  flex-grow: 1;
}

.title {
  flex-grow: 1;
  text-align: left;
}

.AddMovieModal {
  max-width: 400px;
  max-height: 800px;
  padding-bottom: 2rem;
  margin: auto;
}

.AddMovieForm > div {
  margin: 0.5rem 2rem;
}

.AddMovieForm > div > div {
  width: 100%;
}

.AddMovieFormButtons {
  float: right;
  margin-left: 0.3rem !important;
}

.ModalContainer {
  padding: 1rem 1rem 3rem 1rem;
}

.closeIcon {
  position: absolute;
  right: 1rem;
  top: 1.3rem;
  cursor: pointer;
}
