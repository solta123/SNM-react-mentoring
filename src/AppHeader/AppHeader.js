import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    title: {
        flexGrow: 1,
        textAlign: 'left'
    },
    addIcon: {
        paddingRight: '8px'
    }
}));

const AppHeader = () => {
    const classes = useStyles();
  
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
              <Typography variant="h6" className={classes.title}>
                  <b>netflix</b>Roulette
              </Typography>
              <Button color="inherit" variant="outlined">
                  <AddIcon></AddIcon>Add movie
              </Button>
          </Toolbar>
        </AppBar>
      </div>
    );
}

export default AppHeader;
