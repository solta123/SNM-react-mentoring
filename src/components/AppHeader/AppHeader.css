.root {
  flex-grow: 1;
}

.title {
  flex-grow: 1;
  text-align: left;
  color: white;
  text-decoration: none;
}

.AddMovieModal {
  max-width: 400px;
  max-height: 800px;
  margin: auto;
}

.MuiDialogContent-root {
  padding: 0 !important;
}
