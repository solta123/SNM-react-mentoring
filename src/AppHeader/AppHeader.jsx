import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import { DialogContent } from '@material-ui/core';
import Modal from '@material-ui/core/Modal';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

import './AppHeader.css';
import AddMovieModal from '../AddMovieModal/AddMovieModal';
import { initialState } from '../store/reducers/movie';
import { getMovies } from '../store/actionCreator';
import * as actionTypes from '../store/actions';

export const AppHeader = props => {
  return (
    <div className="root">
      <AppBar position="static">
        <Toolbar>
          <NavLink to="/" className="title" onClick={props.getMovies}>
            <Typography variant="h6">
              <b>netflix</b>Roulette
            </Typography>
          </NavLink>
          <Button color="inherit" variant="outlined" onClick={() => props.handleModal(true)}>
            <AddIcon></AddIcon>Add movie
          </Button>
          <Modal open={props.isAddModalOpen} className="AddMovieModal">
            <DialogContent>
              <AddMovieModal onCloseModal={() => props.handleModal(false)} />
            </DialogContent>
          </Modal>
        </Toolbar>
      </AppBar>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    isAddModalOpen: state.movie.isAddModalOpen
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getMovies: () => dispatch(getMovies({ ...initialState, search: '' })),
    handleModal: value => dispatch({ type: actionTypes.MODAL, value: value }),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppHeader);
