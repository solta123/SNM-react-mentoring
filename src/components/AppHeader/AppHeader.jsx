import './AppHeader.css';
import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import AddMovieModal from '../AddMovieModal/AddMovieModal';
import Modal from '@material-ui/core/Modal';
import { DialogContent, Select, MenuItem, FormControl } from '@material-ui/core';
import { NavLink } from 'react-router-dom';
import { setToDefaultState } from '../store/actionCreator';
import { connect } from 'react-redux';
import * as actionTypes from '../store/actions';
import PropTypes from 'prop-types';
import { useTranslation } from "react-i18next";

const AppHeader = props => {
    const { t, i18n } = useTranslation('common');
    const tSource = 'NETFLIXROULETTE.APP_HEADER.';

  const handleLangChange = event => {
    i18n.changeLanguage(event.target.value);
    props.handleLang(event.target.value);
  };

  return (
    <div className="root">
      <AppBar position="static">
        <Toolbar>
          <NavLink to="/" className="title" onClick={props.getMovies}>
            <Typography variant="h6">
              <b>netflix</b>Roulette
            </Typography>
          </NavLink>
          <Button id="AddMovieButton" color="inherit" variant="outlined" onClick={() => props.handleModal(true)}>
            <AddIcon></AddIcon>{t(tSource + 'ADD_MOVIE')}
          </Button>
          <FormControl id="lang-button">
            <Select id="lang-select" value={props.lang} onChange={handleLangChange}>
              <MenuItem id="en" value="en">en</MenuItem>
              <MenuItem id="hu" value="hu">hu</MenuItem>
            </Select>
          </FormControl>
          <Modal open={props.isAddModalOpen} className="AddMovieModal">
            <DialogContent>
              <AddMovieModal onCloseModal={() => props.handleModal(false)} />
            </DialogContent>
          </Modal>
        </Toolbar>
      </AppBar>
    </div>
  );
}

AppHeader.propTypes = {
  isAddModalOpen: PropTypes.bool,
  handleModal: PropTypes.func,
  getMovies: PropTypes.func,
  lang: PropTypes.string,
  handleLang: PropTypes.func
}

const mapStateToProps = state => {
  return {
    isAddModalOpen: state.movie.isAddModalOpen,
    lang: state.movie.lang
  }
}

const mapDispatchToProps = dispatch => {
  return {
    getMovies: () => dispatch(setToDefaultState()),
    handleModal: value => dispatch({ type: actionTypes.MODAL, value }),
    handleLang: lang => dispatch({ type: actionTypes.LANG, lang })
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppHeader);
