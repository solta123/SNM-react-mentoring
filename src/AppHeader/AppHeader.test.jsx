import React from 'react';
import { mount } from "enzyme";
import createTestStore from "../test/testHelper";
import { Provider } from "react-redux";
import AppHeader from './AppHeader';
import { Router, Switch } from 'react-router';

describe("AppHeader test", () => {
    let store;
    beforeEach(() => {
        store = createTestStore();
    });

    it('renders correctly', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router>
                    <AppHeader />
                </Router>
            </Provider>);
        expect(wrapper).toMatchSnapshot();
    });
});