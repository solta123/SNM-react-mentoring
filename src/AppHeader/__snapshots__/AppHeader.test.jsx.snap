// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader test renders correctly 1`] = `ReactWrapper {}`;
